pkg/controller/association/controller/beat_kibana.go

```diff
@@ -68,12 +72,40 @@ func getBeatKibanaRoles(associated commonv1.Associated) (string, error) {
 		)
 	}
 
+	if strings.Contains(beat.Spec.Type, ",") {
+		return "", fmt.Errorf("beat type %s should not contain a comma", beat.Spec.Type)
```

Comment:
If it's a general thing about the Beats CRD it should probably live outside the association controller, and be also enforced by the Beat controller & webhook?

Reply:
It's already forbidden by the Type regex requirements. One here is just double checking. Do you think it would be worth to check it in a validation too?

Reply:
If it's in the OpenAPI validation it should never get through, so I think we can omit the check

Reply:
We seem to have an [issue](https://github.com/elastic/cloud-on-k8s/issues/3290) with that. But even without it, I'd like to keep another check close to the code just to make sure we are safe in case it'll get copied over somewhere not behind that regex check.

Reply:
Ah if the OpenAPI validation doesn't work we should still just check for it in the validation code we run at the beginning of the reconcile/webhook IMO rather than having type validation code scattered about